plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/domains/Injection.kt

```diff
@@ -40,4 +41,34 @@ class Injection(
             }
         }
     }
-}
\ No newline at end of file
+}
+
+
+/**
+ * Viper function that operates on the images of an injection.
+ *
+ * For example, if `original` Viper function operates on `Int` and returns `Bool`,
+ * then resulting `InjectionImageFunction` will take `Ref` of type `intType()` as an argument
+ * and return `Ref` of type `boolType()`.
+ *
+ * @param argsInjections injections that must be applied to the arguments of the operation
+ * @param resultInjection injection that must be applied to the result of the operation
+ * @param checkDivisor adds precondition that divisor (second argument) is not zero
+ */
+class InjectionImageFunction(
+    name: String,
+    val original: Applicable,
+    argsInjections: List<Injection>,
+    resultInjection: Injection,
+    checkDivisor: Boolean = false
```

Comment:
Instead of this, you could add an optional "hook" that lets you add more preconditions.